ui/app/AppLayouts/Wallet/popups/swap/SwapSignModal.qml

```diff
@@ -0,0 +1,247 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+import AppLayouts.Wallet.panels 1.0
+import AppLayouts.Wallet.popups 1.0
+
+import utils 1.0
+
+SignTransactionModalBase {
+    id: root
+
+    required property string fromTokenSymbol
+    required property string fromTokenAmount
+    required property string fromTokenContractAddress
+
+    required property string toTokenSymbol
+    required property string toTokenAmount
+    required property string toTokenContractAddress
+
+    required property string accountName
+    required property string accountAddress
+    required property string accountEmoji
+    required property string accountColorId
+
+    required property string networkShortName // e.g. "oeth"
+    required property string networkName // e.g. "Optimism"
+    required property string networkIconPath // e.g. `Style.svg("network/Network=Optimism")`
+    required property string networkBlockExplorerUrl
+
+    required property string currentCurrency
+    required property string fiatFees
+    required property string cryptoFees
+    required property double slippage
+
+    property string serviceProviderName: "Paraswap"
+    property string serviceProviderURL: "https://www.paraswap.io/"
+
```

Comment:
can you add a TODO to this task here, because its only hardcoded temporarily. https://github.com/status-im/status-desktop/issues/15329

Also only icon and url are harcoded for now, the name comes from the actual backend

Reply:
We don't display the icon here, will add a TODO for the URL

Reply:
TODO added